shared/src/Unison/BlockStore.hs

```diff
@@ -0,0 +1,21 @@
+module Unison.BlockStore where
+
+import Data.ByteString (ByteString)
+
+newtype Hash = Hash ByteString
```

Comment:
@sfultong should delete this newtype and replace all usages of it in this file with the generic `h`